### PR TITLE
fix: drop unique index on mail_messages_snoozed.message_id

### PR DESCRIPTION
**Pull request by JohannesGGE**
Fix: #8748

Followup to #8694

I had to rewrite the snooze db insert. Pls check if the throwing of the ServiceException is correct, I'm not quite sure.

### CODE REVIEW

File: lib/Service/SnoozeService.php [hunk 1]
```diff
@@ -222,20 +218,14 @@ public function snoozeThreadDB(Message $selectedMessage, int $unixTimestamp, Mai
...
-					if($messageId === null) {
-						throw $e;
-					}
-					$this->messageSnoozeMapper->deleteByMessageIds(array($messageId));
-					$this->messageSnoozeMapper->insert($snooze);
-				} else {
-					throw $e;
-				}
+
+			$messageId = $message['messageId'];
```
Comment (kesselb) [inline lib/Service/SnoozeService.php:222]: Why would findMessageIdsByThreadRoot return null for messageId?

File: lib/Service/SnoozeService.php [hunk 2]
```diff
@@ -222,20 +218,14 @@ public function snoozeThreadDB(Message $selectedMessage, int $unixTimestamp, Mai
...
+			if($messageId === null) {
+				throw new ServiceException('Cannot extract messageId from message to snooze.');
 			}
+
+			$this->messageSnoozeMapper->deleteByMessageIds(array($messageId));
```
Comment (kesselb) [inline lib/Service/SnoozeService.php:227]:
```suggestion
			$this->messageSnoozeMapper->deleteByMessageIds([$messageId]);
```

File: lib/Service/SnoozeService.php [hunk 3]
```diff
@@ -167,26 +167,21 @@ public function snoozeThread(
...
-			} else {
-				throw $e;
-			}
+
+		$messageId = $message->getMessageId();
+		if($messageId === null) {
+			throw new ServiceException('Cannot extract messageId from message to snooze.');
 		}
+
+		$this->messageSnoozeMapper->deleteByMessageIds(array($messageId));
```
Comment (kesselb) [inline lib/Service/SnoozeService.php:183]:
```suggestion
		$this->messageSnoozeMapper->deleteByMessageIds([$messageId]);
```

File: lib/Service/SnoozeService.php (hunk 1 above)
Comment (JohannesGGE) [reply lib/Service/SnoozeService.php:223]: You're right, messageId shouldn't be null. I just adapted the pattern from the $message->getMessageId(); where the null check was needed.

**Review by kesselb — APPROVED** [review]

**Pull request merged**